### DISCUSSION TITLE
Benefits

### DISCUSSION THREAD

**Discussion by debuxxed** [headlesshq/headlessmc#73] [Q&A] [answered]
What are benefits of using HMC? I know normal MC takes eats ram drastically, it's java yk, but does HMC in fact help here?

**Comment by 3arthqu4ke** [comment, marked as answer]
> I know normal MC takes eats ram drastically, it's java yk, but does HMC in fact help here?

No

Some things HMC does:
> * add a command line interface to Minecraft, which can control the game.
> * redirect every method in the lwjgl library, causing Minecraft not to render anything, thus making it "headless".
> * patch the Log4J vulnerability.

The main reason I made it was to launch the Minecraft client from console on headless remote servers. E.g. when running a [PingBypass](https://github.com/3arthqu4ke/3arthh4ck#proxypingbypass) server.

It also allows me to deploy the Minecraft client with mods in docker containers.
